File: client/python/example/quick/quick.py
import grpc
import logging

import connect_pb2

from testFinger import testFinger
from testCard import testCard
from testUser import testUser
from testEvent import testEvent

from example.client.client import GrpcClient
from example.connect.connect import ConnectSvc
from example.device.device import DeviceSvc
from example.finger.finger import FingerSvc
from example.card.card import CardSvc
from example.user.user import UserSvc
from example.event.event import EventSvc

CA_FILE = '../../../cert/ca.crt'
SERVER_IP = '192.168.0.2'
SERVER_PORT = 4000

DEVICE_IP = '192.168.0.110'
DEVICE_PORT = 51211
USE_SSL = False


def testConnect(connectSvc):
  try:
    connInfo = connect_pb2.ConnectInfo(IPAddr=DEVICE_IP, port=DEVICE_PORT, useSSL=USE_SSL)

    devID = connectSvc.connect(connInfo)

    devList = connectSvc.getDeviceList()
    print(f'Device list: {devList}', flush=True)

    return devID

  except grpc.RpcError as e:
    print(f'Cannot complete the connect test: {e}')


def testDevice(deviceSvc, deviceID):
  try:
    info = deviceSvc.getInfo(deviceID)

    print(f'Device info: \n{info}', flush=True)

    capInfo = deviceSvc.getCapInfo(deviceID)

    print(f'Capability info: \n{capInfo}', flush=True)

  except grpc.RpcError as e:
    print(f'Cannot complete the device test: {e}')


def quickStart():
  try:
    client = GrpcClient(SERVER_IP, SERVER_PORT, CA_FILE)
    channel = client.getChannel()
    
    connectSvc = ConnectSvc(channel)
    deviceSvc = DeviceSvc(channel)
    fingerSvc = FingerSvc(channel)
    cardSvc = CardSvc(channel)
    userSvc = UserSvc(channel)
    eventSvc = EventSvc(channel)
    
    devID = testConnect(connectSvc)
    testDevice(deviceSvc, devID)
    testFinger(fingerSvc, devID)
    testCard(cardSvc, devID)
    testUser(userSvc, fingerSvc, devID)
    testEvent(eventSvc, devID)

    deviceIDs = [devID]
    connectSvc.disconnect(deviceIDs)

    devList = connectSvc.getDeviceList()
    print(f'\nDevice list: {devList}')

  except grpc.RpcError as e:
    print(f'Cannot run the quick start example: {e}')

if __name__ == '__main__':
    logging.basicConfig()
    quickStart()

File: client/node/example/quick/quick.js
const grpc = require('grpc');
const fs = require('fs');
const connect = require('../connect');
const device = require('../device');
const finger = require('../finger');
const testFinger = require('./testFinger');
const card = require('../card');
const testCard = require('./testCard');
const user = require('../user');
const testUser = require('./testUser');
const event = require('../event');
const testEvent = require('./testEvent');

const CA_FILE = '../../../cert/ca.crt';
const SERVER_IP = '192.168.0.2';
const SERVER_PORT = 4000;

const DEVICE_IP = '192.168.0.110';
const DEVICE_PORT = 51211;
const USE_SSL = false;


function testConnect() {
  var deviceID = 0;

  return connect.connectToDevice(DEVICE_IP, DEVICE_PORT, USE_SSL)
  .then((devID) => {
    deviceID = devID;
    return connect.getDeviceList();
  })
  .then((devList) => {
    console.log('Device list: ', devList);

    return deviceID;
  })
  .catch((err) => {
    console.error('Cannot finish the connect test: ', err);
    throw err;
  });
}



function testDevice(devID) {
  return device.getInfo(devID)
  .then((info) => {
    console.log('Device: ', info);

    return device.getCapabilityInfo(devID)
  })
  .then((capInfo) => {
    console.log('Capability: ', capInfo);
  })
  .catch((err) => {
    console.error('Cannot finish the device test: ', err);
    throw err;
  });
}


function main() {
  var rootCa = fs.readFileSync(CA_FILE);
  var sslCreds = grpc.credentials.createSsl(rootCa);

  connect.initClient(`${SERVER_IP}:${SERVER_PORT}`, sslCreds);
  device.initClient(`${SERVER_IP}:${SERVER_PORT}`, sslCreds);
  finger.initClient(`${SERVER_IP}:${SERVER_PORT}`, sslCreds);
  card.initClient(`${SERVER_IP}:${SERVER_PORT}`, sslCreds);
  user.initClient(`${SERVER_IP}:${SERVER_PORT}`, sslCreds);
  event.initClient(`${SERVER_IP}:${SERVER_PORT}`, sslCreds);

  var deviceID = 0;

  testConnect()
  .then((devID) => {
    deviceID = devID;

    return testDevice(deviceID);
  })
  .then(() => {
    return testFinger.test(deviceID);
  }) 
  .then(() => {
    return testCard.test(deviceID);
  }) 
  .then(() => {
    return testUser.test(deviceID);
  }) 
  .then(() => {
    return testEvent.test(deviceID);
  })
  .then(() => {
    var deviceIDs = [];
    deviceIDs.push(deviceID);

    return connect.disconnect(deviceIDs);
  })
  .then(() => {
    return connect.getDeviceList();
  })
  .then((devList) => {
    console.log('Device list: ', devList);
  })
  .catch((err) => {
    console.error(err);

    if(deviceID != 0) {
      var deviceIDs = [];
      deviceIDs.push(deviceID);
  
      connect.disconnect(deviceIDs);      
    }
  })
}

main();